quiz-app/client/src/main.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+import { RouterProvider } from "react-router-dom";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+import router from "./routers.tsx";
+const queryClient = new QueryClient({
```

Comment:
should create a lib/ folder. lib/react-query . in this folder, we will include many 3rd libs.
need to use queryClient to avoid re-render
```ts
const queryClient = useRef(new QueryClient())
 ```